grammar/haxe.bnf

```diff
@@ -254,11 +254,12 @@ private topLevelDeclaration ::= classDeclaration
 
 privateKeyWord ::= 'private'
 externOrPrivate ::= 'extern' | privateKeyWord
+private externAndMaybePrivate ::= privateKeyWord? 'extern' privateKeyWord?
 
 typedefDeclaration ::= macroClassList? externOrPrivate? 'typedef' componentName genericParam? '=' functionTypeWrapper ';'?
 {pin=5 mixin="com.intellij.plugins.haxe.lang.psi.impl.AbstractHaxeTypeDefImpl" implements="com.intellij.plugins.haxe.lang.psi.HaxeClass"}
 
-externClassDeclaration ::= macroClassList? externOrPrivate* ('class' | 'interface') componentName genericParam? inheritList? '{' externClassDeclarationBody '}'
+externClassDeclaration ::= macroClassList? externAndMaybePrivate ('class' | 'interface') componentName genericParam? inheritList? '{' externClassDeclarationBody '}'
```

Comment:
Don't you want to drop the 'interface' from the or clause here?
If you do, make sure you fix the pin=4 to pin=3 in the line below.